clock/testing/fake_clock_test.go

```diff
@@ -235,13 +235,21 @@ func TestTimerNegative(t *testing.T) {
 	if !tc.HasWaiters() {
 		t.Errorf("expected a waiter to be present, but it is not")
 	}
-	go tc.Step(1 * time.Millisecond)
-	tick := <-timer.C()
+	// force waiters to be called
+	tc.Step(0)
+	tick := assertReadTime(t, timer.C())
 	if tick != tc.Now() {
 		t.Errorf("expected -1s to turn into now: %v != %v", tick, tc.Now())
 	}
 }
 
+func TestTickNegative(t *testing.T) {
+	tc := NewFakeClock(time.Now())
+	if t1, t2 := tc.Tick(-1*time.Second), time.Tick(-1*time.Second); t1 != t2 {
+		t.Errorf("expected negative tick to be the same: %v != %v", t1, t2)
```

Comment:
I think it'd be better to just test the actual contract which is (copying from the doc):

> Tick will return nil if d <= 0.

Reply:
Agreed that would be clearer; done.